### DISCUSSION TITLE
Explanation on LuaRef moveTo

### DISCUSSION THREAD

**Discussion by Mellnik** [kunitoki/LuaBridge3#111] [Q&A] [answered]
In my understanding `LuaRef` is a wrapper around the Lua registry to save values. The registry is global on all lua threads (coroutines).

Just the stack is separate on each thread. So `lua_xmove` just moves a value from one lua thead stack to another.

Could you please explain the usage of `LuaRef::moveTo`?

**Comment by kunitoki** [comment, marked as answer]
You are right, the unref / ref shouldn't be needed, but was not sure if the vm was not doing some bookkeeping of the references in each thread.

**Comment by kunitoki** [comment]
This should have been addressed in latest master
